core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -120,10 +141,14 @@ case class GroupSummary(state: String,
  *  3. leader id
  */
 @nonthreadsafe
-private[coordinator] class GroupMetadata(val groupId: String, val protocolType: String) {
+private[coordinator] class GroupMetadata(val groupId: String) {
 
+  private var state: GroupState = Empty
   private val members = new mutable.HashMap[String, MemberMetadata]
-  private var state: GroupState = Stable
+  private val offsets = new mutable.HashMap[TopicPartition, OffsetAndMetadata]
+  private val pendingOffsetCommits = new mutable.HashMap[TopicPartition, OffsetAndMetadata]
+
+  var protocolType: Option[String] = None
```

Comment:
while the `protocols` may change, `protocolType` shouldn't really change. So It's surprising that this is a var. Is this for groups migrating to new consumer or something?